app/uk/gov/hmrc/transitmovements/utils/SmallMessageLimit.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.utils
+
+import uk.gov.hmrc.transitmovements.config.AppConfig
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SmallMessageLimit @Inject() (config: AppConfig) {
+  private lazy val limit = config.smallMessageSizeLimit
+
+  def checkContentSize(size: Long): Boolean = size match {
+    case (x) if x > limit => true
+    case (_)              => false
+  }
```

Comment:
This is a bit overkill for what you're doing here - it boils down to

```scala
def checkContentSize(size: Long): Boolean = size > limit
```

Does this warrant a separate class like this? I imagine you've done this because of the injection of `AppConfig`. At most, I'd probably make this a `trait` in the `controllers` package, possibly something like

```scala
trait SmallMessageLimit {

    def config: AppConfig // supply this in the controller constructor

    private lazy val limit = config.smallMessageSizeLimit

    def checkContentSize(size: Long): Boolean = size > limit
}
```

which would still be testable by doing

```
object Harness extends SmallMessageLimit {
    override val config: AppConfig = ??? // dummy impl here
}

Harness.checkContentSize(123)
```

That said, if you want to keep this as a separate class, then because you're injecting things, make this the `SmallMessageLimitService` and put it in the `services` package.

Reply:
Continued with the service option